grpc-dump/dump/dump_interceptor.go

```diff
@@ -31,6 +57,7 @@ func dumpInterceptor(logger logrus.FieldLogger, output io.Writer, decoder proto_
 
 		fullMethod := strings.Split(info.FullMethod, "/")
 		md, _ := metadata.FromIncomingContext(ss.Context())
+		dss.Lock()
```

Comment:
Just because I think there's less risk of deadlock if an early return gets added below at some point in future.
```suggestion
		dss.Lock()
		defer dss.Unlock()
```